dips/copy_to_netx.py

```diff
@@ -74,27 +81,38 @@ def parse_component_id_from_mets(mets_data, namespaces):
     This logic should be made more robust in the future.
 
     """
-    # First: attempt to find it as the Dublin Core identifier
-    """
-    namespaces = {
-        "mets": "http://www.loc.gov/METS/",
-        "dcterms": "http://purl.org/dc/terms/",
-        "dc": "http://purl.org/dc/elements/1.1/"
-    }
-    """
+    # First: attempt to find it as the SIP-level Dublin Core identifier
+    objects_div = mets_data.xpath(
+        "/mets:mets/mets:structMap/mets:div/mets:div[@LABEL='objects']",
+        namespaces=namespaces
+    )
+
+    if len(objects_div):
+        # Use DMD ID find correct Dublin Core identifier
+        dmd_id = objects_div[0].get('DMDID')
```

Comment:
The DMDID attribute can actually contain multiple DMD ids, it's not very intuitive but correct in terms of the standard. When multiple DMD sections are associated to the same element, you should try to find the latest one.

See how we fixed this in AtoM:

https://github.com/artefactual/atom/commit/4764110f9426d4fc288b2e64bd185a7fb3d0c053

And how it's being done in SCOPE (not really the best way as it queries for and orders all DMD sections and then filters, instead of only querying for the ones you want and sort):

https://github.com/CCA-Public/scope/blob/master/scope/parsemets.py#L301-L309

SCOPE makes sure that DC metadata exists in the DMD sections. This is to workaround an specific issue they had with missing metadata in the most recent section. I consider this a workaround because the updated metadata could actually be no metadata (intentionally removing the original metadata), but it may a good addition for this case too.

Reply:
Was the CREATED attribute recently added in AM's METS files? I can't find it in the METS files on the AM testing instance I'm using.

Reply:
It's not added in the original DMD sections, just in the updated ones. You may need to trigger a partial re-ingest and change part of the Dublin Core metadata in the AM GUI to get those updated DMD sections.

Reply:
I talked to Justin and he's okay with the script not having logic to get the component ID from Dublin Core so I've removed it to simplify things (they'll still be able to netx.json to specify the component ID, though).